### ISSUE TITLE
Support for mayflash usb to genesis controller

### ISSUE THREAD

**Issue by filevans** [ekeeke/Genesis-Plus-GX#321]
Hello please could you add support for the mayflash usb to genesis controller, it has 2 genesis controller ports. Thank you very much for listening

**Comment by ekeeke** [comment]
Is there any emulator on Wii supporting this USB adapter with some code available ? I don't own this adapter so even if I added support for USB controllers (which are not supported actually), it would be hard for me to add support for this one in particular without any source code examples.

PR are also always welcome (never happened in 10 years regarding Wii port enhancements but a man can hope ^^)

**Comment by filevans** [comment]
Hello, on SNES 9x GX it supports some usb controllers, maybe it will work? I will test it now to see if it works on that

**Comment by filevans** [comment]
Please could you add support for usb controllers anyway? It may work with that, and at least then you could use a genesis usb pad

**Comment by GB-CG** [comment]
@filevans I'm guessing you've not seen this?: https://www.raphnet-tech.com/products/genesis_to_wii/index.php This may do the job perfectly.

**Comment by filevans** [comment]
Yes I've got one of those now. But the mayflash supports 2 in one so it would be even better

**Comment by AlfredoDore** [comment]
I bought a Sinde Winder, which has the same key mapping as the Sega Saturn, it would be wonderful to work, as the experience would be more immersive (The controller hasn't arrived yet, I'll test it as soon as it arrives)
![Image](https://github.com/user-attachments/assets/6fbc361c-8dc7-4343-9fb3-2121bd3b2c79)

**Comment by AlfredoDore** [comment]
I conducted a more in-depth investigation and even consulted the official Sidewinder documentation (including the Linux implementation), in addition to running more comprehensive tests using Nintendont's HID test, and these are the values ​​I arrived at.
Neutral 3F 3F 00 00
Left 00 3F 00 00
Right 7F 3F 00 00
Up 3F 00 00 00
Down 3F 7F 00 00

Buttons
A 3F 3F 01 00
B 3F 3F 02 00
C 3F 3F 04 00
X 3F 3F 08 00
Y 3F 3F 10 00
Z 3F 3F 20 00
L 3F 3F 40 00
R 3F 3F 80 00
Start 3F 3F 00 01
Mode 3F 3F 00 02

button combinations
A + B 03 A + C 05 A + X 09 A + Y 11 A + Z 21 A + L 41 A + R 81 B + C 06 B + X 0A B + Y 12 B + Z 22 B + L 42 B + R 82 C + X 0C C + Y 14 C + Z 24 C + L 44 C + R 84 X + Y 18 X + Z 28 X + L 48 X + R 88 Z + L 60 Z + R A0 Start + Mode 03
It would be interesting to implement robust USB support to handle various types of USB controllers, including Sega's own.
docs: https://docs.kernel.org/input/joydev/joystick.html
Side Winder pad https://github.com/torvalds/linux/blob/master/drivers/input/joystick/sidewinder.c?utm_source=chatgpt.com
